core/io/resource.h

```diff
@@ -125,6 +125,8 @@ class Resource : public RefCounted {
 
 	Node *get_local_scene() const;
 
+	virtual bool is_fully_loaded() const { return true; }
```

Comment:
Not sure about this approach, in my opinion it has confusing semantics which could potentially lead to introducing bugs. CC @RandomShaper

Note that we would like to do a major refactoring of `GDScriptCache`. Maybe we can simplify it, given that `ResourceLoader` now supports "shallow" references.

Reply:
Could you elaborate on this a bit more? Are there other resources that can be "shallow" loaded?
Would you like for me to rename the function to something more contextual?

Otherwise, as far as I understand the `ResourceLoader` is resource agnostic and has to decide whether it wants to use a cached resource or load it from scratch. So the only 2 points where a possible fix can be made are: 1. When the `ResourceLoader` decides to use or not use a cached resource or 2. When it is critically important for a resource to be fully loaded. Through 2. would mean that this has to be decided for a bunch of different points in the engine one by one.

Reply:
I'm with @dalexeev. If the GDScript cache is going to be overhauled, it's better to think how the new implementation can accommodate the need. That may collaborate with the shallow resource support in `ResourceLoader` or maybe allow to make it a `ResourceCache` feature somehow. And that also goes for `GDScriptCache`: it may also be subsumed in `ResourceCache` in the new approach.

However, we'd stil benefit form a short term solution. I was wondering, if the various calls to `Resource::set_path()` in the GDScript module could be the basis of the solution. The idea would involve using `set_path_cache()` instead in some cases (full logic to be determined by the contributor). That one won't register the resource in the `ResourceCache` and so the `ResourceLoader` won't consider it for returning that one. At some later point, when the script is known to be fully loaded, `set_cache()` would be used.

Reply:
I updated the commit. Could you look over it again?

Reply:
It looks reasonable now, but I can't trust I understand all the implications to GDScript. I'd still like @dalexeev to take another look.